### PR TITLE
Fix config appender registration

### LINKED ISSUE #9817: Filebeat autodiscoverkubernetes appender error bug
Please post all questions and issues on https://discuss.elastic.co/c/beats
before opening a Github Issue. Your questions will reach a wider audience there,
and if we confirm that there is a bug, then you can open a new issue.

For security vulnerabilities please only send reports to security@elastic.co.
See https://www.elastic.co/community/security for more information.

Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: filebeat:6.5.4 or filebeat-oss:6.5.3
- Operating System:  centos 7.4
- Discuss Forum URL:  https://discuss.elastic.co/t/filebeat-autodiscoverkubernetes/161967
- Steps to Reproduce: 

reference Filebeat kubernetes autodiscover Multiple conditions 1

The problem is When the condition has “kubernetes.labels.log-index” ，fields:log_topic: 'labels-{data.kubernetes.labels.log-index}', otherwise fields: log_topic: 'aip-{data.kubernetes.namespace}-${data.kubernetes.pod.name}'
The default condition not.equals.kubernetes.container.name: "filebeat" 。

Currently in use appenders ，But error is ERROR	instance/beat.go:691	Exiting: unknown autodiscover appender config
Reference : Filebeat autodiscovery appenders , But there are still ERROR instance/beat.go:691	Exiting: unknown

filebeat run kubernetes daemonset.
filebeat docker image
docker.elastic.co/beats/filebeat:6.5.4
or
docker.elastic.co/beats/filebeat-oss:6.5.3

filebeat.yaml

 ```
filebeat.autodiscover:
    providers:
      - type: kubernetes
        hints.enabled: true
        templates:
          - condition.and:
              - not.has_fields: ['kubernetes.labels.log-index']
              - not.equals.kubernetes.container.name: "filebeat"
            config:
              - type: docker
                containers.ids:
                  - "${data.kubernetes.container.id}"
                exclude_lines: ["^\\s+[\\-`('.|_]"]  # drop asciiart lines
                fields:
                  log_topic: 'aip-${data.kubernetes.namespace}-${data.kubernetes.pod.name}'
        appenders:
          - type: config
            condition.and:                      
              - has_fields: ['kubernetes.labels.log-index']
              - not.equals.kubernetes.container.name: "filebeat"        
            config:
              fields:
                log_topic: 'labels-${data.kubernetes.labels.log-index}'
```

### PR DESCRIPTION
**Pull request by exekias**
Autodiscover config wasn't regsistered, this change updates the code,
adds integration tests and makes sure that the appender is correctly
registered and working.

closes elastic/beats#9817

### CODE REVIEW

**Comment by urso** [comment]
Unfortunately metricbeat CI failed on some appender related tests: 

```
> ERROR: Test config appenders correctly updates configs
```

File: libbeat/autodiscover/appenders/config/config.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ package config
 import (
 	"fmt"
 
+	"github.com/pkg/errors"
```
Comment (urso) [inline libbeat/autodiscover/appenders/config/config.go]: Did you mean "errors" or "github.com/pkg/errors". Just asking as I'm seeing this wrong import a lot recently.

Comment (exekias) [reply libbeat/autodiscover/appenders/config/config.go]: https://github.com/pkg/errors is the one that gives me `errors.Wrap`, I think this is correct?

**Comment by urso** [comment]
Please backport to 6.6 branch as well.

**Pull request merged**